src/openrct2/platform/Platform.Posix.cpp

```diff
@@ -319,19 +317,18 @@ namespace Platform
     bool EnsureDirectoryExists(u8string_view path)
     {
         mode_t mask = openrct2_getumask();
-        char buffer[MAX_PATH];
-        SafeStrCpy(buffer, u8string(path).c_str(), sizeof(buffer));
+        u8string buffer(path);
 
-        LOG_VERBOSE("Create directory: %s", buffer);
-        for (char* p = buffer + 1; *p != '\0'; p++)
+        LOG_VERBOSE("Create directory: %s", buffer.c_str());
+        for (utf8* p = buffer.data() + 1; *p != '\0'; p++)
         {
             if (*p == '/')
             {
                 // Temporarily truncate
                 *p = '\0';
```

Comment:
This is all levels of dangerous. This violates the u8string contract. Before when it was just a buffer it was allowable but messy. This function shouldn't even exist anymore. We can use std::filesystem create_directory and exists.

Reply:
What kind of contract? `std::basic_string` is allowed to have null terminators inside it if I recall correctly. That said, I agree this code is meh.

Reply:
Oh yeah looks like string does allow multiple nulls still hate it. Yeah replace with filesystem